rest_framework_json_api/utils.py

```diff
@@ -304,6 +304,21 @@ def get_resource_type_from_serializer(serializer):
     )
 
 
+def get_resource_id_from_instance(resource_instance, resource):
+    """Returns the resource identifier for a given instance (`id` takes priority over `pk`)."""
+    if resource and "id" in resource:
+        return resource["id"] and encoding.force_str(resource["id"]) or None
+    if resource_instance:
+        if hasattr(resource_instance, "get_id") and callable(resource_instance.get_id):
```

Comment:
What is the use case to have a `get_id` method on the resource instance resp. model? The model is the database representation, whereas the `id` is the JSON:API spec representation. So if the `id` needs to be different from the database it should be defined on the serializer by adding a `id` field. That should be enough as far as I see it.